packages/replay-next/src/suspense/FrameCache.ts

```diff
@@ -37,3 +37,40 @@ export function getFrameSuspense(
   const frames = getFramesSuspense(replayClient, pauseId);
   return frames?.find(frame => frame.frameId === frameId);
 }
+
+export const {
+  getValueSuspense: getTopFrameSuspense,
+  getValueAsync: getTopFrameAsync,
+  getValueIfCached: getTopFrameIfCached,
+} = createGenericCache<
+  [replayClient: ReplayClientInterface],
+  [pauseId: PauseId],
+  Frame | undefined
+>(
+  "FrameCache: getTopFrame",
+  1,
+  async (client, pauseId) => {
+    // In most cases, we probably already have a full set of frames cached for this pause ID.
+    // Try to use the first frame from there if possible.
+    const existingCachedFrames = getFramesIfCached(pauseId);
+
+    if (existingCachedFrames?.value) {
+      return existingCachedFrames.value[0];
+    }
+
+    // Otherwise, we'll use a lighter-weight `Pause.getTopFrame` request. The object
+    // that comes back _should_ be the same either way.
+    const framesResult = await client.getTopFrame(pauseId);
+    const { frame: frameId } = framesResult;
+
+    if (frameId === undefined) {
+      return;
+    }
+    await client.waitForLoadedSources();
+    cachePauseData(client, pauseId, framesResult.data);
+    const topFrame = framesResult.data?.frames?.find(frame => frame.frameId === frameId);
```

Comment:
The frame still needs to have its location(s) updated (to account for corresponding locations), like [here](https://github.com/replayio/devtools/blob/f929de2bb80b9facf629571f0f6bcafa8fe4215b/packages/replay-next/src/suspense/PauseCache.ts#L120-L126)